topi/include/topi/nn/pooling.h

```diff
@@ -285,6 +289,158 @@ inline Tensor global_pool(const Tensor& x,
   }
 }
 
+inline Expr start_index(Expr index, Expr odim, Expr idim) {
+  return tvm::floor((index * idim) / odim);
+}
+
+inline Expr end_index(Expr index, Expr odim, Expr idim) {
+  return tvm::ceil((index + 1) * idim / odim);
+}
+
+/*!
+* \brief Perform adaptive pooling on height and width dimension of data.
+*
+* \param x The input tensor
+* \param out_height Output height
+* \param out_width Output width
+* \param pool_type The type of pooling operator
+* \param height_axis index of the height dimension
+* \param width_axis index of the width dimension
+*
+* \return The output tensor in same layout order
+*/
+inline Tensor adaptive_pool_impl(const Tensor& x,
+                                 const size_t out_height,
+                                 const size_t out_width,
+                                 PoolType pool_type,
+                                 const size_t height_axis,
+                                 const size_t width_axis) {
+  CHECK(x->shape.size() == 4) << "Adaptive Pooling input must be NCHW.";
+  CHECK(out_height > 0) << "Output height must be positive.";
+  CHECK(out_width > 0) << "Output width must be positive.";
+
+  Expr height = tvm::ir::Simplify(x->shape[height_axis]);
+  Expr width = tvm::ir::Simplify(x->shape[width_axis]);
+
+  tvm::Map<tvm::Array<tvm::Expr>, tvm::IterVar> dheights;
```

Comment:
Here's an attempted solution using a map to get precomputed `IterVar`s. Feedback please @kevinthesun @Laurawly @icemelon9 @yuruofeifei

Reply:
I get the following error message in unit test:
```
terminate called after throwing an instance of 'std::out_of_range'
  what():  _Map_base::at
Aborted (core dumped)
```
Could it be possible that it's attempting to access the map with `Array<Var>` instead of `Array<Expr>`?